src/quorum/datadriven_test.rs

```diff
@@ -123,7 +123,76 @@ fn test_quorum(data: &TestData) -> String {
+
+                let idx = c.committed_index(use_group_commit, &l);
+                buf.push_str(&c.describe(&l));
+
+                // Joining a majority with the empty majority should give same result.
+                let a_idx =
+                    JointConfig::new_joint_from_majorities(c.clone(), MajorityConfig::default())
+                        .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via zero-joint quorum\n", a_idx.0));
+                }
+
+                // Joining a majority with itself should give same result.
+                let a_idx = JointConfig::new_joint_from_majorities(c.clone(), c.clone())
+                    .committed_index(use_group_commit, &l);
+                if a_idx != idx {
+                    buf.push_str(&format!("{} <-- via self-joint quorum\n", a_idx.0));
+                }
+
+                // test overlaying
+                for &id in c.ids() {
+                    if let Some(iidx) = l.acked_index(id) {
+                        if idx.0 > iidx.index {
+                            // try index - 1
+                            l.insert(
+                                id,
+                                Index {
+                                    index: iidx.index - 1,
+                                    group_id: iidx.group_id,
+                                },
+                            );
+
+                            let a_idx = c.committed_index(use_group_commit, &l);
+                            if a_idx != idx {
+                                buf.push_str(&format!(
+                                    "{} <-- overlaying {}->{}\n",
+                                    a_idx.0,
+                                    id,
+                                    iidx.index - 1
+                                ));
+                            }
+                            // try 0
+                            l.insert(
+                                id,
+                                Index {
+                                    index: 0,
+                                    group_id: iidx.group_id,
+                                },
+                            );
+
+                            let a_idx = c.committed_index(use_group_commit, &l);
+                            if a_idx != idx {
+                                buf.push_str(&format!(
+                                    "{} <-- overlaying {}->{}\n",
+                                    a_idx.0, id, 0
+                                ));
+                            }
+                            // recovery
+                            l.insert(id, iidx);
+                        }
+                    }
+                }
+                buf.push_str(&format!(
+                    "{}\n",
+                    Index {
+                        index: idx.0,
+                        group_id: 0
+                    }
+                ));
```

Comment:
Could we just 
```suggestion
                buf.push_str(&format!("{}\n", idx.0));
```

Reply:
When index is MAX,  we need to display `oo`, so we better print in `Index`

Reply:
But the joint quorum branch just prints `idx.0` should we unify them?

Reply:
ok, I will unify them.